Refactor Shape.Rectangle.RotateToIsometry

reaperrr opened the pull request:
Also fixes TS wall and gate shapes.

reaperrr committed:
Refactor Shape.Rectangle.RotateToIsometry

And fixes TS wall and gate shapes.

Phrohdoh commented on OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs:
```diff
@@ -600,6 +600,21 @@ internal static void UpgradeActorRules(ModData modData, int engineVersion, ref L
+					if (node.Key.StartsWith("RotateToIsometry", StringComparison.Ordinal))
+					{
+						var value = FieldLoader.GetValue<string>("RotateToIsometry", node.Value.Value);
+						if (value == "true")
```
You'll need use `value.ToLower()` and compare against `true` and `yes`.

reaperrr commented:
Updated.

pchote commented on OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs:
```diff
@@ -601,6 +601,21 @@ internal static void UpgradeActorRules(ModData modData, int engineVersion, ref L
+				{
+					if (node.Key.StartsWith("RotateToIsometry", StringComparison.Ordinal))
+					{
+						var value = FieldLoader.GetValue<string>("RotateToIsometry", node.Value.Value).ToLower();
```
`GetValue<bool>`, and then you avoid duplicating FieldLoader's boolean parsing.

reaperrr commented:
Updated.

aydjay commented on OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs:
```diff
@@ -601,6 +601,21 @@ internal static void UpgradeActorRules(ModData modData, int engineVersion, ref L
+						if (value == true)
+							node.Value.Value = "128";
+						else
+							node.Value.Value = "0";
```
You may want to use a ternary here for legibility.

'node.Value.Value = value ? "128" : "0";'

pchote commented on OpenRA.Mods.Common/HitShapes/Rectangle.cs:
```diff
@@ -33,10 +33,9 @@ public class RectangleShape : IHitShape
+		[Desc("Rotates shape by an angle relative to actor facing. Mostly required for buildings on isometric terrain.",
 			"Mobile actors do NOT need this!")]
-		public readonly bool RotateToIsometry = false;
+		public readonly WAngle HorizontalRotation = WAngle.Zero;
```
Please call this `LocalYaw` for consistency with our other yaw angle offsets.

reaperrr commented:
Updated, greatly condensed `var orientation` in the hitshape by using WRot.FromYaw.
Still works as before according to my testing, but requesting @pchote to throw this a last glance.

pchote commented:
My :+1: still holds.

reaperrr commented:
Depends on #13253.

reaperrr committed:
Adapt blocking height of TS walls etc.

to WDist rescaling.

Also, make wall hit-shape height match blocking height and tweak it to
approximate original TS behaviour as much as possible (Tick Tanks could
shoot over them, but only from certain angles/distances).

reaperrr committed:
Remove TS infantry hit-shape height

We should either implement this for all non-blocker actors at once, or not at all. By now we have walls/gates as examples for vertical shape height, so for now there's no reason to leave this on infantry.

reaperrr commented:
Rebased and updated to account for #13253.

pchote commented:
My :+1: *still* still holds.

I noticed a couple of things that could be fixed here as a scope creep, or left to a future PR:
* The blocking height of walls and gates are different.
* Component towers and laser fence posts are parts of walls, so should have the same hitshape.
